openshift/machine-config-operator
# OCPBUGS-21721: install/0000_90_machine-config-operator_90_deletion: Drop this file

ace637f50b (#3740, [OCPBUGS-10924](https://issues.redhat.com/browse/OCPBUGS-10924)) moved the 4.14 machine-config operator to a non-default ServiceAccount and ClusterRoleBinding.  But 4.13 and earlier remain on the default ServiceAccount.

1cdb75f6ce (#3923, [OCPBUGS-19400](https://issues.redhat.com/browse/OCPBUGS-19400)) brought `Recreate` logic back to [4.13.14][1] and later (good), but also brought back a `delete` manifest for the default ClusterRoleBinding, which leads to the 4.13 cluster-version operator fighting with itself over whether that ClusterRoleBinding should exist (it should exist on 4.13, [OCPBUGS-21721](https://issues.redhat.com/browse/OCPBUGS-21721)).  For example, [this CI run][2] updates from 4.12.36 to 4.13.14, and has:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-upgrade/1707415968109563904/artifacts/e2e-aws-upgrade/clusterversion.json | jq -r '.items[].status.conditions[] | select(.type == "Upgradeable") | .lastTransitionTime + " " + .type + "=" + .status + " " + .reason + ": " + .message'
2023-09-28T17:09:41Z Upgradeable=False ResourceDeletesInProgress: Cluster minor level upgrades are not allowed while resource deletions are in progress; resources=clusterrolebinding "default-account-openshift-machine-config-operator"
```

**- What I did**

By dropping the deletion manifest from 4.13, we avoid contention between two manifests, and leave the default ClusterRoleBinding alone until a later update to 4.14 will remove it.

**- How to verify it**

Update from 4.12 to 4.13.  Confirm that the default ClusterRoleBinding exists and that ClusterVersion's `Upgradeable` is not complaining about `resource deletions are in progress`.

**- Description for the changelog**

Are we doing an MCO change-log?  I'd expect this to get handled via Jira fields.

[1]:  https://amd64.ocp.releases.ci.openshift.org/releasestream/4-stable/release/4.13.14
[2]: https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-upgrade/1707415968109563904
